src/pjpersist/datamanager.py

```diff
@@ -785,7 +785,16 @@ def abort(self, transaction):
         self._report_stats()
         try:
             if self._tpc_activated:
-                self._conn.tpc_rollback()
+                try:
+                    self._conn.tpc_rollback()
+                except psycopg2.errors.UndefinedObject:
```

Comment:
What kind of object is undefined? Why do you think it is correct to intercept this exception here? Is there any docs prescribing such behaviour?

Reply:
@kedder The tests are demonstrating th eissue.

Reply:
@strichter I don't see the answer to this question in the test

Reply:
Well, it does not really matter what psycopg2 throws at this stage other than that it throws this error when the transaction is already broken.

Reply:
It does matter to understand what is happening. Fixes like this then bring down the whole app. When postgres raises an error while executing the transaction, transaction should be explicitly aborted. Now, why the transaction fails to abort exactly? What happens if we don't abort transactions?